language-detection-fasttext/src/main/java/com/hpe/caf/languagedetection/fasttext/FastTextWrapper.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015-2019 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.languagedetection.fasttext;
+
+import com.hpe.caf.languagedetection.DetectedLanguage;
+import static com.hpe.caf.languagedetection.fasttext.FastTextDetector.MIN_RELIABILITY;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import jep.JepException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FastTextWrapper {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FastTextWrapper.class);
+    private static final double MIN_LINE_ACCURACY = 0.95;
+    private static final int WORDS_PER_SENTENCE = 3;
+    private final String langPrefix = "__label__";
+    
+    public LDResult detect(final String document) {
+        final List<String> lines = new ArrayList<>();
+        for (final String line : document.split("\n")) {
+            lines.add(line.trim());
+        }
+        return lines.size() > 0 ? detect(lines) : new LDResult(new ArrayList(), 0);
+    }
+    
+    private List<String> sptlitTextIntoSentences(final String text) {
+        final List<String> sentences = new ArrayList();
+        final String[] words = text.split(" ");
+        for (int i = 0; i < words.length; i+=WORDS_PER_SENTENCE) {
+            String sentence = "";
+            for (int j = 0; j < WORDS_PER_SENTENCE; j++) {
+                final int index = i+j;
+                if (index < words.length) {
+                    final String word = words[index].trim();
+                    if (!word.isEmpty()) {
+                        sentence = sentence + (j == 0 ? "" : " ") + word;
```

Comment:
StringBuilder, https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java

Reply:
After changes not relevant anymore, but good point as I wasn't aware that between iterations is string concatenation much more expensive the way I did.